src/tool_getparam.c

```diff
@@ -2750,13 +2751,7 @@ ParameterError getparameter(const char *flag, /* f or -long-flag */
       break;
     case C_HELP: /* --help */
       if(toggle) {
-        if(*nextarg) {
-          global->help_category = strdup(nextarg);
-          if(!global->help_category) {
-            err = PARAM_NO_MEM;
-            break;
-          }
-        }
+        tool_help((nextarg && *nextarg) ? nextarg : NULL);
```

Comment:
I've only glanced the code, but wouldn't tool_help cope with just `tool_help(nextarg);` here?

Reply:
That was my first take but it doesn't work. In this case the code has been friendly to make nextarg point to a zero length arg "", which then `tool_help` takes as a blank category. I can of course change the logic and move this into tool_help() instead but I figure it doesn't really matter where it gets done.